lms/services/user.py

```diff
@@ -67,7 +67,6 @@ def _find_existing_user(self, model_user: User) -> Optional[User]:
             .filter_by(
                 application_instance=model_user.application_instance,
                 user_id=model_user.user_id,
-                h_userid=model_user.h_userid,
```

Comment:
I don't see any reason why we need to include `h_userid` in this query: just `(application_instance, user_id)` is unique.

The query originates from https://github.com/hypothesis/lms/pull/3282 but there's no discussion of why `h_userid` was included.

Note that before https://github.com/hypothesis/lms/pull/3494 (merged earlier today) there was no unique constraint on `(application_instance, user_id)`, only on `(application_instance, user_id, h_userid)`. So `(application_instance, user_id)` wasn't guaranteed to be unique. In practice though `(application_instance, user_id)` was actually unique apart from one duplicate row that somehow got into the DB (see <https://hypothes-is.slack.com/archives/C1MA4E9B9/p1643209011047900>).

Also, [`UserService.get()`](https://github.com/hypothesis/lms/blob/c2979ab714bd229e6454adada093a4d58c4a02af/lms/services/user.py#L41-L62) assumes that `(application_instance, user_id)` is unique and will crash otherwise.

Reply:
I think we could do some refactoring here because `get()` and `_find_existing_user()` are almost the same thing and don't both need to exist. Maybe for a follow-up PR though, this one fixes the bug